pkg/server/backend_manager.go

```diff
@@ -214,42 +225,40 @@ func containIDType(idTypes []header.IdentifierType, idType header.IdentifierType
 }
 
 // AddBackend adds a backend.
-func (s *DefaultBackendStorage) AddBackend(identifier string, idType header.IdentifierType, conn agent.AgentService_ConnectServer) Backend {
+func (s *DefaultBackendStorage) AddBackend(identifier string, idType header.IdentifierType, backend *Backend) {
```

Comment:
since we are removing the return value and it seems like there is at least one error condition, should we be returning some sort of error from this function?

Reply:
Similar to @tallclair 's comment about ProxyServer and BackendManager separation, I don't disagree but prefer to keep out of this PR scope.

Reply:
makes sense to me. happy to see the change in a followup PR.